test/blackbox/common/mesh_event_handler.c

```diff
@@ -271,3 +271,22 @@ void mesh_event_destroy(void) {
 	pthread_cancel(event_receive_thread);
 }
 
+bool change_state(node_status_t *status, mesh_event_t currentEv) {
+
+	if(status->status) {
```

Comment:
As I said before, is the `status->status` variable actually needed? What if we write:
```
if(status->current_index == status->max_events) {
    return false;
}
```
Wouldn't that have the same effect? Then the if statement on line 284 can be removed completely, as well as the `bool status` from `node_status_t`.

Reply:
Yes it will have the same effect, but in every test case there is a condition as 
```
        if((node_status[0].status == true) && (node_status[3].status == true)
	                && (node_status[1].status == true) && (node_status[2].status == true)
	                && (node_status[4].status == true) && (node_status[5].status == true)) {
		test_case_status = true;
		return true;
	}

	return false;
```
The latest change will just make the if() condition even more lengthy